test-common/src/main/scala/org/scalajs/testcommon/RPCCore.scala

```diff
@@ -13,19 +13,33 @@ import java.util.concurrent.atomic.AtomicLong
 import Serializer.{serialize, deserialize}
 import FutureUtil._
 
-private[scalajs] abstract class RPCCore {
+
+/** Core RPC dispatcher.
+ *
+ *  Tracks and assigns call identities on top of a message passing interface.
+ *
+ *  Note that it does not have timeout handling for calls. Users are expected to
+ *  manage call teardown by calling [[close]] in case of failure. This typically
+ *  means that subclasses need to put an explicit call to [[close]] once they
+ *  are sure to not call [[handleMessage]] anymore.
+ *
+ *  This class guarantees that dispatch handles synchronously when
+ *  [[handleMessage]] is called, so closing can be performed race-free.
+ */
+private[scalajs] abstract class RPCCore(implicit ex: ExecutionContext) {
```

Comment:
Due to a peculiarity of Scala, classes always need a non-implicit parameter list. This means that this signature in fact desugards to `RPCCore()(implicit ex: ExecutionContext)`. IMO it's better to explicitly write the `()` in that case, to make it obvious.